lib/mpl_toolkits/axes_grid1/inset_locator.py

```diff
@@ -386,42 +389,25 @@ def inset_axes(parent_axes, width, height, loc='upper right',
         Inset axes object created.
     """
 
-    if axes_class is None:
-        axes_class = HostAxes
-    if axes_kwargs is None:
-        axes_kwargs = {}
-    inset_axes = axes_class(parent_axes.figure, parent_axes.get_position(),
-                            **axes_kwargs)
-
-    if bbox_transform in [parent_axes.transAxes,
-                          parent_axes.figure.transFigure]:
-        if bbox_to_anchor is None:
-            _api.warn_external("Using the axes or figure transform requires a "
-                               "bounding box in the respective coordinates. "
-                               "Using bbox_to_anchor=(0, 0, 1, 1) now.")
-            bbox_to_anchor = (0, 0, 1, 1)
-
+    if (bbox_transform in [parent_axes.transAxes, parent_axes.figure.transFigure]
+            and bbox_to_anchor is None):
+        _api.warn_external("Using the axes or figure transform requires a "
+                           "bounding box in the respective coordinates. "
+                           "Using bbox_to_anchor=(0, 0, 1, 1) now.")
+        bbox_to_anchor = (0, 0, 1, 1)
     if bbox_to_anchor is None:
         bbox_to_anchor = parent_axes.bbox
-
     if (isinstance(bbox_to_anchor, tuple) and
             (isinstance(width, str) or isinstance(height, str))):
         if len(bbox_to_anchor) != 4:
             raise ValueError("Using relative units for width or height "
                              "requires to provide a 4-tuple or a "
                              "`Bbox` instance to `bbox_to_anchor.")
-
-    axes_locator = AnchoredSizeLocator(bbox_to_anchor,
-                                       width, height,
-                                       loc=loc,
-                                       bbox_transform=bbox_transform,
-                                       borderpad=borderpad)
-
-    inset_axes.set_axes_locator(axes_locator)
-
-    _add_inset_axes(parent_axes, inset_axes)
-
-    return inset_axes
+    kwargs = {**(axes_kwargs if axes_kwargs is not None else {}),
```

Comment:
Either
```suggestion
    axes_kwargs = {**(axes_kwargs if axes_kwargs is not None else {}),
```
to match the name in the signature; or inline into the function call

Reply:
Tweaked to something in that spirit.